riscv-runtime/src/std_support.rs

```diff
@@ -14,30 +14,36 @@ use powdr_riscv_syscalls::Syscall;
 use crate::io::write_slice;
 use getrandom::{register_custom_getrandom, Error};
 
+/// The std interface to random number generation.
 #[no_mangle]
 extern "C" fn sys_rand(buf: *mut u32, words: usize) {
-    // This is only used by std to key the hash function of HashMap. Until we
-    // figure a way to make random values available to user programs, let's just
-    // have deterministic HashMaps.
-    const VALUE: u32 = 4;
-
-    let slice = unsafe { slice::from_raw_parts_mut(buf, words) };
-    for v in slice.iter_mut() {
-        *v = VALUE;
+    let buf = buf as *mut u8;
+    unsafe {
+        let slice = slice::from_raw_parts_mut(buf, words * 4);
+        powdr_getrandom_impl(slice);
     }
 }
 
-fn powdr_getrandom(s: &mut [u8]) -> Result<(), Error> {
-    // TODO: getrandom expects this u8 slice as input, so we're not using sys_rand.
-    // Both should probably be fixed/unified later once we have a proper RNG.
-    const VALUE: u8 = 3;
-    s.iter_mut().for_each(|v| *v = VALUE);
-
+/// The de-factor standard rust interface to low level random number generation.
```

Comment:
de facto
but i think this comment belongs to `sys_rand`?

Reply:
No, sys_rand is risc0 invention.

getrandom is the defacto standard, it is even used inside std in some targets